app_bikes/forms.py

```diff
@@ -0,0 +1,26 @@
+from dal import autocomplete
+from django import forms
+
+
+class BikeModelForm(forms.ModelForm):
+    def __init__(self, *args, **kw):
+        super(BikeModelForm, self).__init__(*args, **kw)
+        if self.instance is not None:
+            # Saved instance is loaded, setup choices to display the selected value
+            self.fields['id_station'].widget.choices = ((self.instance.id_station, self.instance.station),)
```

Comment:
This was the tricky part and I'm not very statisfied of it. I chose to store id_station, which will be the invariant identifier of the station, and its name (station).
The station field has no other purpose than to be in the JSON output if we want to display the station (in case we have several).
This field is base on a SelectField, which has a predefined list of choices. I could have called the webservice to get all the (id, name) pairs for all the stations of all the providers but it's very time consuming, so to avoid that I have to overwrite the choices:
- when the user inputs something (the autocomplete will be triggered)
- when the user wants to edit the model (this case), I rebuild the choices with only one value, so the autocomplete will display the current station (that's also why "station" field is not displayed in the form)
- In case there was an error in the form, to set its value back to what was selected.

tldr; : it's the second case in your question, hope my answer was clear